pkg/apis/elasticsearch/v1/elasticsearch_types_test.go

```diff
@@ -314,3 +316,88 @@ func TestElasticsearch_DisabledPredicates(t *testing.T) {
 		})
 	}
 }
+
+// Test_AssociationConfs tests that if something resets the AssocConfs map, then AssociationConf() reinitializes
+// the map from the annotation.
+func Test_AssociationConfs(t *testing.T) {
+	// simple es without associations
+	es := &Elasticsearch{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "es",
+			Namespace: "default",
+		},
+	}
+
+	// set assoc conf even if no assoc
+	assert.Equal(t, 0, len(es.AssocConfs))
+	for _, association := range es.GetAssociations() {
+		assocConf, err := assocutils.GetAssociationConf(association)
+		assert.NoError(t, err)
+		association.SetAssociationConf(assocConf)
+	}
```

Comment:
nit: should we just remove this loop and ensure that there is no item returned ?
```suggestion
	assert.Equal(t, 0, len(es.GetAssociations()))
```